java/vector/src/main/codegen/templates/DenseUnionVector.java

```diff
@@ -714,8 +714,22 @@ public int getBufferSizeFor(final int count) {
     if (count == 0) {
       return 0;
     }
+
+    int[] legCounts = new int[typeFields.length];
```

Comment:
What is a "leg" here? This is not common terminology in Arrow.

Reply:
Could call this `typeCounts` or `childCounts`.

Reply:
Forgive the name 😓, internal team term leakage! 

Fixed by adapting setChildVectorValueCounts (using `counts`)